src/rust/engine/src/nodes.rs

```diff
@@ -423,34 +437,80 @@ impl ExecuteProcess {
       timeout: Duration::from_millis((timeout_in_seconds * 1000.0) as u64),
       description: description,
       jdk_home: jdk_home,
-    }))
+    })
+  }
+
+  fn lift(value: &Value) -> Result<MultiPlatformExecuteProcess, String> {
+    let mut request_by_constraint: BTreeMap<(Platform, Platform), ExecuteProcessRequest> =
+      BTreeMap::new();
+    let constraint_parts = externs::project_multi_strs(&value, "platform_constraints");
+    let mut requests: Vec<ExecuteProcessRequest> = Vec::new();
+    for execute_process in externs::project_multi(&value, "execute_process_requests").iter() {
+      let underlying_req = MultiPlatformExecuteProcess::lift_execute_process(execute_process)?;
+      requests.push(underlying_req);
+    }
+    if constraint_parts.len() % 2 != 0 {
+      return Err("Error parsing platform_constraints: odd number of parts".to_owned());
+    }
+    if constraint_parts.len() / 2 != requests.len() {
+      return Err(format!(
+        "Size of constraint keys and requests does not match: {} vs. {}",
+        constraint_parts.len() / 2,
+        requests.len()
+      ));
+    }
+    for (constraint_key_pair, request) in constraint_parts.chunks_exact(2).zip(requests.iter()) {
+      let mut platform_constaint: Vec<Platform> = constraint_key_pair
+        .iter()
+        .map(|c| MultiPlatformExecuteProcess::get_platform_variant(c.to_string()))
```

Comment:
A couple of things about this line...
1. Why is this `MultiPlatformExecuteProcess::get_platform_variant` rather than `Platform::from` or `Platform::try_from`?
2. You don't actually need ownership of the string to do the conversion; you should be able to just pass `c` (or maybe `&c`), change the signature of `get_platform_variant` to take a `&String` (or better yet, you can actually just take a `&str`), and remove the call to `.as_ref()` in `get_platform_variant`